File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/03-additional/06.weddingSeats.js
function weddingSeats(input) {

    let lastSector = input[0];
    let rowsInSectorA = Number(input[1]);
    let seatsCountInOddRow = Number(input[2]);

    let firstSector = 'A';
    let firstSectorAsNum = firstSector.charCodeAt();
    let lastSectorAsNum = lastSector.charCodeAt();
    let seatsCounter = 0;
    let rows = rowsInSectorA;

    for (let i = firstSectorAsNum; i <= lastSectorAsNum; i++) {

        for (let j = 1; j <= rows; j++) {

            let startSeat = 97;
            let seats = '';
            if (j % 2 != 0) {
                seats = seatsCountInOddRow;
            } else {
                seats = seatsCountInOddRow + 2;
            }
            for (let k = 1; k <= seats; k++) {
                seatsCounter++;
                console.log(`${String.fromCharCode(i)}${j}${String.fromCharCode(startSeat)}`);
                startSeat++;
            }
        }
        rows++;
    }

    console.log(seatsCounter);
}
weddingSeats(['C', '4', '2']);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/03.additional/06.wedding_seats.py
last_sector = input()
rows_in_sector_a = int(input())
seats_count_in_odd_row = int(input())

first_sector = 'A'
first_sector_as_num = ord(first_sector)
last_sector_as_num = ord(last_sector)
seats_counter = 0
rows = rows_in_sector_a

for i in range(first_sector_as_num, last_sector_as_num + 1):
    for j in range(1, rows + 1):
        start_seat = 97
        seats = ''
        if j % 2 != 0:
            seats = seats_count_in_odd_row
        else:
            seats = seats_count_in_odd_row + 2
        
        for k in range(1, seats + 1):
            seats_counter += 1
            print(f'{str(chr(i))}{j}{str(chr(start_seat))}')
            start_seat += 1
             
    rows += 1
    
print(seats_counter)
